src/rofiles-fuse/main.c

```diff
@@ -324,13 +374,22 @@ do_open (const char *path, mode_t mode, struct fuse_file_info *finfo)
           return -errno;
         }
 
-      int r = can_write_stbuf (&stbuf);
+      int r = verify_write_or_copyup (path, &stbuf);
       if (r != 0)
         {
           (void) close (fd);
           return r;
         }
 
+      /* In the copyup case, we need to re-open */
+      if (opt_copyup)
+        {
+          (void) close (fd);
+          fd = openat (basefd, path, finfo->flags & ~O_TRUNC, mode);
```

Comment:
Minor: we can just stop filtering out `O_TRUNC` in the copy-up case, right? And only pay the `ftruncate` call in the non-copy-up case.

Reply:
BTW I wish I'd seen this comment while working on this patchset on the plane :airplane:  - it turned out mishandling `O_TRUNC` was the source of bugs.